nexus/db-queries/src/db/datastore/mod.rs

```diff
@@ -137,38 +138,51 @@ pub struct DataStore {
 // to compilation times; changing a query only requires incremental
 // recompilation of that query's module instead of all queries on `DataStore`.
 impl DataStore {
-    /// Constructs a new Datastore object.
+    /// Constructs a new Datastore object, without any version validation.
     ///
-    /// Only returns if the database schema is compatible with Nexus's known
-    /// schema version.
-    pub async fn new(log: &Logger, pool: Arc<Pool>) -> Result<Self, String> {
+    /// Ignores the underlying DB version. Should be used with caution, as usage
+    /// of this method can result in incompatible schemas.
```

Comment:
What does "can result in incompatible schemas" mean?  What I'd expect is something like "this function returns a working DataStore whose methods may read and write data in a way that's incompatible with the way other components are reading and writing to the same database.  Database corruption could result."

Is this only used for updating the DataStore's schema version?  It seems like a (future) safer API would be a separate object for that that only lets you update the schema.

Reply:
I'll update this comment for now. Agreed about the future safe API, but I also dunno how much to optimize for the case, because I think operator-driven updates may not use the CLI tools in the limit.